src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -7678,27 +7690,96 @@ private ExpressionAst MemberAccessRule(ExpressionAst targetExpr, Token operatorT
+                        ?? genericTypeArguments?.LastOrDefault()?.Extent?.EndColumnNumber
+                        ?? member.Extent.EndColumnNumber;
+
                     Diagnostics.Assert(lParen.Kind == TokenKind.LParen || lParen.Kind == TokenKind.LCurly, "token kind incorrect");
-                    Diagnostics.Assert(member.Extent.EndColumnNumber == lParen.Extent.StartColumnNumber,
-                                       "member and paren must be adjacent");
-                    return MemberInvokeRule(targetExpr, lParen, operatorToken, member);
+                    Diagnostics.Assert(
+                        endColumnNumber == lParen.Extent.StartColumnNumber,
+                        "member and paren must be adjacent when the method is not generic");
+                    return MemberInvokeRule(targetExpr, lParen, operatorToken, member, genericTypeArguments);
                 }
             }
 
             return new MemberExpressionAst(
-                    ExtentOf(targetExpr, member),
-                    targetExpr,
-                    member,
-                    @static: operatorToken.Kind == TokenKind.ColonColon,
-                    nullConditional: operatorToken.Kind == TokenKind.QuestionDot);
+                ExtentOf(targetExpr, member),
+                targetExpr,
+                member,
+                @static: operatorToken.Kind == TokenKind.ColonColon,
+                nullConditional: operatorToken.Kind == TokenKind.QuestionDot,
+                genericTypeArguments);
+        }
+
+        private List<ITypeName> GenericMethodTypeArgumentsRule(out Token rBracketToken)
+        {
+            List<ITypeName> genericTypes = null;
+
+            int resyncIndex = _tokenizer.GetRestorePoint();
+            Token lBracket = NextToken();
+            rBracketToken = null;
+
+            if (lBracket.Kind == TokenKind.LBracket)
+            {
+                // This is either a member expression with generic type arguments, or some sort of collection index
+                // on a property.
+                var oldTokenizerMode = _tokenizer.Mode;
+                try
+                {
+                    // Switch to typename mode to avoid aggressive argument tokenization.
+                    SetTokenizerMode(TokenizerMode.TypeName);
+
+                    SkipNewlines();
+                    Token firstToken = NextToken();
+                    if (firstToken.Kind != TokenKind.Number
+                        && (firstToken.Kind == TokenKind.Identifier || firstToken.Kind == TokenKind.LBracket))
+                    {
+                        resyncIndex = -1;
+                        genericTypes = GenericArgumentsRule(firstToken, false, out rBracketToken);
+
+                        if (rBracketToken.Kind != TokenKind.RBracket)
+                        {
+                            UngetToken(rBracketToken);
+                            ReportIncompleteInput(
```

Comment:
If we can, we should test for this case

Reply:
Absolutely! Not sure how to test for this, though. 😅